foundations/foundation-metrics/src/main/java/org/apache/servicecomb/foundation/metrics/meter/BucketsDistribution.java

```diff
@@ -0,0 +1,162 @@
+  private static final Logger LOGGER = LoggerFactory.getLogger(BucketsDistribution.class);
+
+  public static final String METRICS_LATENCY = DynamicPropertyFactory.getInstance()
+      .getStringProperty("servicecomb.metrics.invocation.latency.buckets", "0")
+      .get();
+
+  private List<LatencyInterval> latencyIntervals;
+
+  public BucketsDistribution(Id id) {
+    this.id = id;
+    this.latencyIntervals = parseLatencyIntervalListFromProperty(METRICS_LATENCY, id);
+  }
+
+  public void record(long nanoAmount) {
+    if (nanoAmount > 0) {
+      for (LatencyInterval latencyInterval : latencyIntervals) {
+        if (latencyInterval.match(nanoAmount)) {
+          latencyInterval.increment();
+          return;
+        }
+      }
+    }
+  }
+
+  @Override
+  public void calcMeasurements(long msNow, long secondInterval) {
+    List<Measurement> measurements = new ArrayList<>();
+    calcMeasurements(measurements, msNow, secondInterval);
+    allMeasurements = measurements;
+  }
+
+  @Override
+  public void calcMeasurements(List<Measurement> measurements, long msNow, long secondInterval) {
+    latencyIntervals.forEach(latencyInterval -> {
+      measurements.add(newMeasurement(latencyInterval.getId(),
+          msNow, latencyInterval.getAndUpdateLastTimes()));
+    });
+  }
+
+  private Measurement newMeasurement(Id id, long msNow, Number n) {
+    return new Measurement(id, msNow, n.doubleValue());
+  }
+
+  public static List<LatencyInterval> parseLatencyIntervalListFromProperty(String property, Id rootId) {
+    List<LatencyInterval> results = new ArrayList<>();
+    try {
+      int[] arrays = Arrays.stream(property.trim().split(","))
+          .mapToInt(Integer::parseInt)
+          .distinct()
+          .sorted()
+          .filter(num -> num > 0)
+          .toArray();
+      if (arrays == null || arrays.length == 0) {
+        return results;
+      }
+      results.add(new LatencyInterval(0L, arrays[0], rootId));
+      for (int i = 1; i < arrays.length; i++) {
+        results.add(new LatencyInterval(arrays[i - 1], arrays[i], rootId));
+      }
+      results.add(new LatencyInterval(arrays[arrays.length - 1], Long.MAX_VALUE, rootId));
+    } catch (Throwable e) {
+      LOGGER.error("Failed to parse property servicecomb.metrics.invocation.latency.buckets", e);
+    }
+    return results;
+  }
+
+  public static class LatencyInterval {
+
+    public static final int leastLength = 7;
+
+    private Id id;
+
+    private LongAdder times = new LongAdder();
+
+    private long lastUpdated = 0L;
+
+    private long minValue;
+
+    private long maxValue;
+
+    public LatencyInterval(long minValue, long maxValue, Id rootId) {
+      this.minValue = TimeUnit.MILLISECONDS.toNanos(minValue);
+      this.maxValue = maxValue == Long.MAX_VALUE ? Long.MAX_VALUE : TimeUnit.MILLISECONDS.toNanos(maxValue);
+      this.id = rootId.withTag("statistic", getStandardLatencyStr(minValue, maxValue, leastLength));
```

Comment:
tag no need to same to output format.....
tag is "[0,1)", and output format is "[0,1)  ", seems not conflict, right?

Reply:
I think so, I will redesign the method

Reply:
```java
  private static Id createLatencyScopeId(String leftStr, String rightStr, Id rootId) {
    return rootId.withTag("statistic", LatencyScope.getStandardLatencyScopeStr(leftStr, rightStr, 0));
  }

    public static String getStandardLatencyScopeStr(String leftStr, String rightStr, int minLength) {
      StringBuilder sb = new StringBuilder();
      sb.append("[").append(leftStr)
          .append(",")
          .append(rightStr)
          .append(")");
      if (sb.length() < minLength) {
        for (int i = 0; i < minLength - sb.length(); i++) {
          sb.append(' ');
        }
      }
      return sb.toString();
    }

```